nemo/collections/asr/models/classification_models.py

```diff
@@ -147,112 +325,115 @@ def transcribe(self, paths2audio_files: List[str], batch_size: int = 4, logprobs
-            pin_memory=config.get('pin_memory', False),
+        output2 = os.path.join(os.path.dirname(output), 'decoder_' + os.path.basename(output))
+        output2_descr = qual_name + ' Decoder exported to ONNX'
+        decoder_onnx = self.decoder.export(
+            output2,
+            None,  # computed by input_example()
+            None,
+            verbose,
+            export_params,
+            do_constant_folding,
+            keep_initializers_as_inputs,
+            onnx_opset_version,
+            try_script,
+            set_eval,
+            check_trace,
+            use_dynamic_axes,
         )
 
-    def setup_training_data(self, train_data_config: Optional[Union[DictConfig, Dict]]):
-        if 'shuffle' not in train_data_config:
-            train_data_config['shuffle'] = True
-        # preserve config
-        self._update_dataset_config(dataset_name='train', config=train_data_config)
+        output_model = attach_onnx_to_onnx(encoder_onnx, decoder_onnx, "EDC")
+        output_descr = qual_name + ' Encoder+Decoder exported to ONNX'
+        onnx.save(output_model, output)
+        return ([output, output1, output2], [output_descr, output1_descr, output2_descr])
 
-        self._train_dl = self._setup_dataloader_from_config(config=train_data_config)
 
-        # Need to set this because if using an IterableDataset, the length of the dataloader is the total number
-        # of samples rather than the number of batches, and this messes up the tqdm progress bar.
-        # So we set the number of steps manually (to the correct number) to fix this.
-        if 'is_tarred' in train_data_config and train_data_config['is_tarred']:
-            # We also need to check if limit_train_batches is already set.
-            # If it's an int, we assume that the user has set it to something sane, i.e. <= # training batches,
-            # and don't change it. Otherwise, adjust batches accordingly if it's a float (including 1.0).
-            if isinstance(self._trainer.limit_train_batches, float):
-                self._trainer.limit_train_batches = int(
-                    self._trainer.limit_train_batches
-                    * ceil((len(self._train_dl.dataset) / self.world_size) / train_data_config['batch_size'])
-                )
+class EncDecClassificationModel(_EncDecBaseModel):
+    """Encoder decoder Classification models."""
 
-    def setup_validation_data(self, val_data_config: Optional[Union[DictConfig, Dict]]):
-        if 'shuffle' not in val_data_config:
-            val_data_config['shuffle'] = False
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
 
-        # preserve config
-        self._update_dataset_config(dataset_name='validation', config=val_data_config)
+        super().__init__(cfg=cfg, trainer=trainer)
 
-        self._validation_dl = self._setup_dataloader_from_config(config=val_data_config)
+        # Change labels if needed
+        self._update_decoder_config(self._cfg.decoder)
```

Comment:
This should be done *before* calling `super.__init__` so that the decoder config is updated properly. Otherwise, let baseclass call it *before* it calls the `_setup_decoder`.